cfme/containers/image.py

```diff
@@ -138,6 +139,88 @@ def all(self):
                                            provider=provider or get_crud_by_name(ems_name)))
         return images
 
+    def check_compliance_multiple_images(self, image_entities, check_on_entity=True, timeout=240):
+        """Initiates compliance check and waits for it to finish on several Images.
+        Args:
+            image_entities: list of Image entities that need to perform compliance check on them
+            check_on_entity (bool): check the compliance status on the entity summary view if True,
+                                    only run compliance otherwise.
+            timeout (seconds): time for waiting for compliance status
+        """
+
+        # Chose Check Compliance of Last Known Configuration
+        images_view = navigate_to(self, 'All')
+        self.check_image_entities(image_entities)
+        wait_for(lambda: images_view.toolbar.policy.is_enabled, num_sec=5,
+                 message='Policy drop down menu is disabled after checking some Images')
+        images_view.toolbar.policy.item_select('Check Compliance of Last Known Configuration',
+                                  handle_alert=True)
+        images_view.flash.assert_no_error()
+
+        # Verify Image summary
+        if check_on_entity:
+            for image_instance in image_entities:
+                original_state = 'never verified'
+                try:
+                    wait_for(
+                        lambda: image_instance.compliance_status.lower() != original_state,
+                        num_sec=timeout, delay=5,
+                        message='compliance state of Image ID, "{}", still matches {}'
+                                .format(image_instance.id, original_state)
+                    )
+                except TimedOutError:
+                    logger.error('compliance state of Image ID, "{}", is {}'
+                                 .format(image_instance.id, image_instance.compliance_status))
+                    raise TimedOutError('Timeout exceeded, Waited too much'
+                                        ' time for check Compliance finish ({}).'.format(timeout))
+
+    def check_image_entities(self, image_entities):
+        """check rows on Container Images table."""
+        images_view = navigate_to(self, 'All', use_resetter=False)
```

Comment:
Checkout the `apply` method on `EntitesConditionalView`, there's a few places here where I think it would be useful.

https://github.com/ManageIQ/integration_tests/blob/master/widgetastic_manageiq/__init__.py#L3065

Reply:
I am familiar with this apply method, I opened Jira ticket on izapolsk in order to improve my test performance.
Since my PR is on review status for long time, and the apply method wasn't merged, I didn't use it.
According to your comment i use the apply method on Image.py and cfme/common/__init__.py.